GSoC/DMP 2025 Week 1: feat: add brick SVG path generation logic

saumyashahi opened the pull request:
## Description
This PR adds an `index.ts` file implementing the SVG path generation logic for rendering different brick types using dynamic path calculation functions in the masonry module.

## Key Features 

- `generatePath()` main function to generate SVG path strings for different brick types:
           - Simple bricks (type1)
           - Expression bricks (type2)
           - Compound bricks (type3)
           
- Support for edge-wise path construction:
           - Top: Includes optional notch and label width adjustment
           - Right: Dynamically handles multiple argument notches 
           - Bottom: Adjusts based on label width and bricky type
           - Left: Includes optional connection notch
- `getBoundingBox()` function added to calculate bBox for each brick

saumyashahi committed:
feat(masonry): add brick SVG path generation logic

justin212407 committed:
refactor(masonry): modularize path generator and fix lint issues

Signed-off-by: Justin Charles <charlesjustin2124@gmail.com>

meganindya commented:
next PR onwards please follow the commit pattern: `<type>(masonry): [scope inside masonry] <message>`

meganindya commented:
Also move this file to `src/masonry/brick/utils/path.ts`. You'll be adding a `spec/path.spec.ts` within the same utils folder.

saumyashahi committed:
refactor(masonry): move path generator file to utils folder

saumyashahi commented:
> next PR onwards please follow the commit pattern: `<type>(masonry): [scope inside masonry] <message>`

I fixed it, I rebased the commits, to reword the commit messages itself keeping the history clean. I will keep in mind from the next commits.

saumyashahi commented:
> Also move this file to `src/masonry/brick/utils/path.ts`. You'll be adding a `spec/path.spec.ts` within the same utils folder.

I have moved the path.ts file to `/brick/utils/path.ts`

justin212407 committed:
feat(masonry): standardize SVG origin in path

Signed-off-by: Justin Charles <charlesjustin2124@gmail.com>

justin212407 committed:
test(masonry): Add unit test for path file

Signed-off-by: Justin Charles <charlesjustin2124@gmail.com>

justin212407 committed:
chore(Masonry): fix lint errors in path.spec.ts

Signed-off-by: Justin Charles <charlesjustin2124@gmail.com>

saumyashahi committed:
feat(masonry): add bounding box calculation logic in path.ts
